go-controller/pkg/kube/kube.go

```diff
@@ -139,3 +141,29 @@ func (k *Kube) GetNamespaces() (*kapi.NamespaceList, error) {
 func (k *Kube) GetNetworkPolicies(namespace string) (*kapisnetworking.NetworkPolicyList, error) {
 	return k.KClient.Networking().NetworkPolicies(namespace).List(metav1.ListOptions{})
 }
+
+func (k *Kube) GetConfigFromNetworkAttachment(namespace string, networkName string) (map[string]interface{}, error) {
+	rawPath := fmt.Sprintf("/apis/k8s.cni.cncf.io/v1/namespaces/%s/network-attachment-definitions/%s", namespace, networkName)
+	netData, err := k.KClient.ExtensionsV1beta1().RESTClient().Get().AbsPath(rawPath).DoRaw()
+	if err != nil {
+		return nil, err
+	}
+
+	customResource := &NetworkAttachmentDefinition{}
+	if err := json.Unmarshal(netData, customResource); err != nil {
+		return nil, err
+	}
+
+	emptySpec := NetworkAttachmentDefinitionSpec{}
+	if customResource.Spec == emptySpec {
```

Comment:
is this different than checking that Spec.Config == "" ?

Reply:
No, seems it is the same.
But I copied my code from https://github.com/intel/multus-cni/blob/67e661f932a6aba869275a0a4d7db865fea888ae/k8sclient/k8sclient.go#L312 so maybe we should stay consistent with them.

Reply:
if it is the same than please use the simpler format (instead of creating an empty object)